### DISCUSSION TITLE
New hardware Thin Client Wyse Z90D7

### DISCUSSION THREAD

**Discussion by defdefred** [VendelinSlezak/BleskOS#45] [Show and tell]
https://www.parkytowers.me.uk/thin/wyse/z/z90d7/

Using a multi partitioned usb flash drive, I was able to see the bleskOS bootloader, but booting give only a black screen...

How can I troubleshoot?

**Comment by VendelinSlezak** [comment]
At first try to use USB only with BleskOS image, without any partitions.

**Comment by defdefred** [comment]
Same black screen of death.

**Comment by VendelinSlezak** [comment]
If update 7 will produce same result, please open in bootloader [a] Boot options>[g] Select graphic mode and post screen here.

**Comment by defdefred** [comment]
I can browse all graphic mode in [a] and testing several gives always the black screen.
Could you produce debug message to file on usb partition 2 ?
Or maybe some pc speaker beep to find the failing step of graphical init by counting the beep?

**Comment by VendelinSlezak** [comment]
Okay, here you have image that have BleskOS bootloader that is exactly same as normal, except that at end of booting process it do not jump to execute BleskOS code, but it redraws part of screen. By this we will be able to locate if problem occurs in bootloader itself, or problem is in C code.

[bleskos_test_bootloader_without_bleskos_c_code_img.zip](https://github.com/VendelinSlezak/BleskOS/files/13906505/bleskos_test_bootloader_without_bleskos_c_code_img.zip)

Result should be something like this, or green part of screen:

![Snímka obrazovky_2024-01-11_18-28-45](https://github.com/VendelinSlezak/BleskOS/assets/43180618/80f01f9d-96e3-44ae-a8bc-fbf164f19ab7)

**Comment by defdefred** [comment]
I have the green top of screen.

**Comment by VendelinSlezak** [comment]
I maybe found where mistake happend, can you please test this version?

[bleskos_2024u8_test_of_vesa_img.zip](https://github.com/VendelinSlezak/BleskOS/files/13927439/bleskos_2024u8_test_of_vesa_img.zip)

**Comment by defdefred** [comment]
I have the same top green screen.

**Comment by VendelinSlezak** [comment]
From this last image [bleskos_2024u8_test_of_vesa_img.zip](https://github.com/VendelinSlezak/BleskOS/files/13927439/bleskos_2024u8_test_of_vesa_img.zip) ?

**Comment by defdefred** [comment]
Yep
```
  489  cd /home/wize/Téléchargements/
  490  unzip bleskos_2024u8_test_of_vesa_img.zip 
  491  dd if=bleskos_2024u8_test_of_vesa.img skip=1 of=/dev/sdb seek=1
```

**Comment by VendelinSlezak** [comment]
Then something must not be working with dd command (maybe sudo privileges are missing?) because if you will run this image in emulator, there is nothing about green top of screen, it is normal Update 8 only with little change in vesa.c file.

**Reply by defdefred** [reply]
Is the file uploaded to github valid?
I run dd as root the same way I test my code.

**Reply by VendelinSlezak** [reply]
I downloaded it and tested it so it should be valid. Maybe you can try another USB?

**Reply by defdefred** [reply]
With another usb key I have a black screen.

**Comment by VendelinSlezak** [comment]
Okay, here you have bootloader that will show only values about mode that was selected, so we will be able to see, if problem is that BleskOS is reading wrong values.

[bleskos_bootloader_print_mode_info_img.zip](https://github.com/VendelinSlezak/BleskOS/files/13940305/bleskos_bootloader_print_mode_info_img.zip)

![Snímka obrazovky_2024-01-15_16-07-32](https://github.com/VendelinSlezak/BleskOS/assets/43180618/197dae00-9b20-487e-b88a-1d1ef0991e33)

**Comment by defdefred** [comment]
![20240115_193646](https://github.com/VendelinSlezak/BleskOS/assets/17281367/99b3eacd-4c1e-4de5-854f-9b22043286f1)
Et Voilà

**Reply by VendelinSlezak** [reply]
I do not see image, would you please upload it again?

**Reply by defdefred** [reply]
I edited the msg

**Comment by VendelinSlezak** [comment]
Thank you. It looks like problem is not there. I prepared image that at beginning of C code draws yellow top of screen, and after initalizing memory whole blue screen so we will be able to find out where it stucks.

[bleskos_2024u9_colors_during_start_img.zip](https://github.com/VendelinSlezak/BleskOS/files/13942263/bleskos_2024u9_colors_during_start_img.zip)

**Comment by defdefred** [comment]
I have a thick line of yellow on top.

**Comment by VendelinSlezak** [comment]
Nice, this specifies more that problem happens in first few functions. Can you please test this image and tell result?

[bleskos_2024u9_colors_during_start_2_img.zip](https://github.com/VendelinSlezak/BleskOS/files/13942406/bleskos_2024u9_colors_during_start_2_img.zip)

**Comment by defdefred** [comment]
blue top

**Comment by VendelinSlezak** [comment]
Great. So initalize_memory() works. If initalize_logging() works too, this image should show green top.

[bleskos_2024u9_colors_during_start_3_img.zip](https://github.com/VendelinSlezak/BleskOS/files/13942511/bleskos_2024u9_colors_during_start_3_img.zip)

**Reply by defdefred** [reply]
yes green top it is

**Comment by VendelinSlezak** [comment]
I added option of debug messages into booting. In boot options, turn off manual selecting of graphic mode and turn on debug messages and then start BleskOS with special boot options. It should change color of top of screen after first methods and then after inializing graphic it show logs, so we will precisely know where it stucks.

[bleskos_2024u10_test_version_img.zip](https://github.com/VendelinSlezak/BleskOS/files/13942632/bleskos_2024u10_test_version_img.zip)

![Snímka obrazovky_2024-01-15_20-59-56](https://github.com/VendelinSlezak/BleskOS/assets/43180618/73fbe39d-3ea8-46c2-a970-f7f9c72463e2)

**Comment by defdefred** [comment]
Red screen of death 
Informations about graphic mode can not be readed

**Comment by VendelinSlezak** [comment]
Did you turn off manual selection of graphic mode? Or maybe I messed it up and turning it off is in fact on. Could you please test both options? I am now going to sleep so I can not check it now, but I will continue to work on this problem tomorrow.

**Comment by defdefred** [comment]
Red screen is when I use [a]

**Comment by defdefred** [comment]
seems again a dd issue

**Comment by defdefred** [comment]
I only have ligth blue top

**Comment by defdefred** [comment]
need to sleep early sorry.

**Comment by VendelinSlezak** [comment]
We are closer to finding out what exactly is wrong. Can you please try Update 10 with debug messages on?

**Comment by defdefred** [comment]
white top

**Comment by VendelinSlezak** [comment]
Thank you, this locates issue to initalize_scalable_font() Can you please try if Update 11 works?

**Comment by defdefred** [comment]
white (grey) top

**Comment by VendelinSlezak** [comment]
Huh. I am really wondering what causes this problem. Can you please test this image? 
[bleskos_2024u12_debug_29_01_img.zip](https://github.com/VendelinSlezak/BleskOS/files/14088788/bleskos_2024u12_debug_29_01_img.zip)

**Comment by defdefred** [comment]
white (grey) top again.

**Comment by VendelinSlezak** [comment]
Would you please try this version? I hope it will show at least full white screen.

[bleskos_2024u15_debug_version_img.zip](https://github.com/VendelinSlezak/BleskOS/files/14168776/bleskos_2024u15_debug_version_img.zip)

**Comment by defdefred** [comment]
Full red screen!

**Comment by VendelinSlezak** [comment]
Nice. I prepared three different tests to speed up this testing.

[bleskos_2024u15_test_1_img.zip](https://github.com/VendelinSlezak/BleskOS/files/14230217/bleskos_2024u15_test_1_img.zip)
[bleskos_2024u15_test_2_img.zip](https://github.com/VendelinSlezak/BleskOS/files/14230216/bleskos_2024u15_test_2_img.zip)
[bleskos_2024u15_test_3_img.zip](https://github.com/VendelinSlezak/BleskOS/files/14230215/bleskos_2024u15_test_3_img.zip)

**Comment by defdefred** [comment]
1:boot to main window but no mouse nor keyboard
2:not booting
3:not booting

I tested both on another pc with success.

**Comment by defdefred** [comment]
Is it the expected result?

**Comment by VendelinSlezak** [comment]
Well, no, expected result is booting all three images :) Would you please try to boot first image with debug option on?

**Comment by defdefred** [comment]
Haha! It was an issue with the usb hub embedded in the HP screen!
[wizethinclient.txt](https://github.com/VendelinSlezak/BleskOS/files/14337030/wizethinclient.txt)

**Comment by VendelinSlezak** [comment]
That's great. Thank you for testing BleskOS on this computer.
